playbooks/deploy-phobos-nodes.yml

```diff
@@ -55,8 +56,23 @@
         meta: "instance_role=ceph-{{ cluster_deploy_version }}-haproxy"
         timeout: 720
         validate_certs: false
+        cloud: phobos
       register: haproxy_nodes
 
+    - name: Create Client Server Group
+      os_server_group:
+        state: present
+        name: "ceph-{{ cluster_deploy_version }}-clients"
+        policies: "anti-affinity"
+        validate_certs: false
+        cloud: phobos
+      register: client_server_group
+      when: client_count > 0
+
+    - name: Set Scheduler Hint Fact
+      set_fact:
+        scheduler_hint: "group={{ client_server_group.id | default('') }}"
```

Comment:
you could probably apply the when clause from above instead of defaulting '' but that doesn't matter too much.